gi_core/dispatch/math/vectorized/gamma_index_range_impl.cpp

```diff
@@ -0,0 +1,594 @@
+#include <math/vectorized/gamma_index_range_impl.hpp>
+
+#include <simdpp/dispatch/get_arch_raw_cpuid.h>
+#define SIMDPP_USER_ARCH_INFO ::simdpp::get_arch_raw_cpuid()
+
+namespace yagit::core::math::vectorized::detail
+{
+	namespace SIMDPP_ARCH_NAMESPACE
+	{
+		// -------- float32 --------
+
+		constexpr size_t float32_VectorSize = SIMDPP_FAST_FLOAT32_SIZE;
+
+		error_code gamma_index_initialize_pass(view<float> gamma_index_output, view<float> gamma_index_output_end)
+		{
+			if (is_aligned_to<float32_VectorSize>(gamma_index_output))
+			{
+				return detail::gamma_index_initialize_pass<true, float32_VectorSize>(gamma_index_output, gamma_index_output_end);
+			}
+			else
+			{
+				return detail::gamma_index_initialize_pass<false, float32_VectorSize>(gamma_index_output, gamma_index_output_end);
+			}
+		}
+
+		error_code gamma_index_minimize_pass(
```

Comment:
This and the following `gamma_index_minimize_pass` blocks of code seem to be almost identical.
Explain in comment how they differ and that is the reason of such code duplication

Reply:
this is a dynamic dispatch/multiversioned file (specifically https://p12tic.github.io/libsimdpp/v2.1/libsimdpp/w/arch/dispatch.html), TLDR it gets "dispatched" into several compilation units from which each one gets compiled under different vectorization flags. In the end when running the program SIMDPP_MAKE_DISPATCHER selects a version of this function that is compatible with the instruction set that current CPU supports. Also they are sligthly different (amount of arguments passed and types used)